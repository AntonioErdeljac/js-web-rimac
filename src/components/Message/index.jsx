import PropTypes from 'prop-types';
import React, { memo } from 'react';
import { Alert } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const Message = ({ title, description, icon, variant }) => {
  return (
    <Alert variant={variant} className="text-white mb-4">
      <div className="d-inline-flex align-items-center">
        <FontAwesomeIcon size="lg" icon={icon} />
        <strong className="ml-3">{title}</strong>
      </div>
      <div className="mt-3">{description}</div>
    </Alert>
  );
};

Message.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  variant: PropTypes.string.isRequired,
};

export default memo(Message);
