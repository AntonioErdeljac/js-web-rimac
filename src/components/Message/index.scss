@import '../../scss/bootstrap';

.r-alert {
  width: auto;
  height: auto;

  @media only screen and (max-width: map-get($grid-breakpoints, "lg")) {
    display: none !important;
  }
}
