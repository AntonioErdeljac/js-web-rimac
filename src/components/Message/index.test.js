import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Message from './index';

const makeProps = ({ description, icon, variant } = {}) => ({
  description: description || 'All systems operational',
  icon: icon || 'exclamation-circle',
  variant: variant || 'success',
});

const renderComponent = (props) => shallow(<Message {...props} />);

describe('Message Component', () => {
  it('renders correctly according to snapshot', () => {
    expect(toJson(renderComponent(makeProps()))).toMatchSnapshot();
  });

  it('renders proper description', () => {
    const expectedDescription = 'Lorem Ipsum';

    const wrapper = renderComponent(makeProps({ description: expectedDescription, icon: null }));
    expect(wrapper.find('.ml-2').text()).toEqual(expectedDescription);
  });
});
