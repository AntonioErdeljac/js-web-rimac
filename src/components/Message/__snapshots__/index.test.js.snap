// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component renders correctly according to snapshot 1`] = `
<Alert
  className="r-alert text-white mb-0 border-0 shadow-sm"
  closeLabel="Close alert"
  show={true}
  transition={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "appear": false,
        "in": false,
        "mountOnEnter": false,
        "timeout": 300,
        "unmountOnExit": false,
      },
      "render": [Function],
    }
  }
  variant="success"
>
  <div
    className="d-inline-flex align-items-center"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="exclamation-circle"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="lg"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <div
      className="ml-2"
    >
      All systems operational
    </div>
  </div>
</Alert>
`;
